components/datetime/src/provider/gregory.rs

```diff
@@ -218,15 +245,29 @@ pub mod patterns {
         feature = "provider_serde",
         derive(serde::Serialize, serde::Deserialize)
     )]
-    pub struct SkeletonsV1(pub LiteMap<SkeletonV1, PatternV1>);
+    pub struct SkeletonsV1<'s>(
+        #[cfg_attr(feature = "provider_serde", serde(borrow))]
+        pub  LiteMap<SkeletonV1, PatternV1<'s>>,
+    );
 
     #[derive(Debug, PartialEq, Clone, Default)]
     #[cfg_attr(
         feature = "provider_serde",
         derive(serde::Serialize, serde::Deserialize)
     )]
-    pub struct DateTimeFormatsV1 {
-        pub length_patterns: LengthPatternsV1,
-        pub skeletons: SkeletonsV1,
+    pub struct DateTimeFormatsV1<'s> {
+        #[cfg_attr(feature = "provider_serde", serde(borrow))]
+        pub length_patterns: LengthPatternsV1<'s>,
+        #[cfg_attr(feature = "provider_serde", serde(borrow))]
+        pub skeletons: SkeletonsV1<'s>,
+    }
+
+    impl<'s> DateTimeFormatsV1<'s> {
+        pub fn zero_copy_from<'b>(this: &'b DateTimeFormatsV1<'s>) -> DateTimeFormatsV1<'b> {
```

Comment:
I suspect I should name it differently, but I can't implement the trait because the struct doesn't implement `Yokable`. Should it?